lib/Map/TableStructure.js

```diff
@@ -503,6 +514,118 @@ TableStructure.prototype.getIdMapping = function() {
     }, {});
 };
 
+/**
+ * Appends table2 to this table. If rowNumbers are provided, only takes those
+ * row numbers from table2.
+ * Changes all the columns in one go, to avoid partial updates from tracked values.
+ * @param {TableStructure} table2 The table to add to this one.
+ * @param {Integer[]} [rowNumbers] The row numbers from table2 to add (defaults to all).
+ */
+TableStructure.prototype.append = function(table2, rowNumbers) {
+    if (this.columns.length !== table2.columns.length) {
+        throw new DeveloperError('Cannot add tables with different numbers of columns.');
+    }
+    var updatedColumnValueses = [];
+    for (var columnNumber = 0; columnNumber < table2.columns.length; columnNumber++) {
+        var valuesToAdd = table2.columns[columnNumber].values;
+        if (defined(rowNumbers)) {
+            valuesToAdd = valuesToAdd.filter((_, rowNumber) => rowNumbers.indexOf(rowNumber) >= 0);
```

Comment:
Right so I remember what I was thinking here - using `filter` is fine, it's `indexOf` that I'm worried about. Every value in `valuesToAdd` will result in a linear search over `rowNumbers`, whereas if you turned `rowNumbers` into a lookup table at the start your filter function could just be `defined(rowNumbersLookup[rowNumber])`